main.go

```diff
@@ -72,54 +85,99 @@ func addGriefer(s *discordgo.Session, m *discordgo.MessageCreate) {
 
 				return
 			}
 		}
 
-		for _, mention := range m.Mentions {
+		for _, mention := range mess.Mentions {
 			griefers = append(griefers, mention.ID)
 		}
 
-		s.ChannelMessageSend(m.ChannelID, "This brotha is getting griefed")
+		sess.ChannelMessageSend(mess.ChannelID, "This brotha is getting griefed")
 
 		return
 	}
 }
 
-func handleCommands(s *discordgo.Session, m *discordgo.MessageCreate) {
-	if m.Author.ID != s.State.User.ID {
-		log.Printf(m.Author.Username + ": " + m.Content)
+func handleCommands(sess *discordgo.Session, mess *discordgo.MessageCreate) {
+	if mess.Author.ID != sess.State.User.ID {
+		log.Printf(mess.Author.Username + ": " + mess.Content)
 	}
 
 	// Respond to messages
-	if m.Content == "!hello" {
+	if mess.Content == "!hello" {
 		// Reply with a message
-		s.ChannelMessageSend(m.ChannelID, "Hello, "+m.Author.Username+"!")
+		sess.ChannelMessageSend(mess.ChannelID, "Hello, "+mess.Author.Username+"!")
 	}
 
-	if m.Content == fmt.Sprintf("<@%v>", vetroId) {
-		s.ChannelMessageSend(m.ChannelID, "Hey, "+m.Author.Username+"!")
+	if mess.Content == fmt.Sprintf("<@%v>", vetroId) {
+		sess.ChannelMessageSend(mess.ChannelID, "Hey, "+mess.Author.Username+"!")
 
 	}
-	if m.Content == "https://imgur.com/a/XQ3pPTQ" {
-		s.ChannelMessageSend(m.ChannelID, "Assemble!!!!!")
+	if mess.Content == "https://imgur.com/a/XQ3pPTQ" {
+		sess.ChannelMessageSend(mess.ChannelID, "Assemble!!!!!")
 	}
 }
 
-func handleVoiceStateUpdate(s *discordgo.Session, m *discordgo.VoiceStateUpdate) {
-	if m.ChannelID != channelId {
+func handleVoiceStateUpdate(sess *discordgo.Session, mess *discordgo.VoiceStateUpdate) {
+	if mess.ChannelID != channelId {
 		return
 	}
 
-	if m.VoiceState.UserID == morthisId {
-		s.ChannelMessageSend(m.ChannelID, fmt.Sprintf("Hello gaylord <@%s>", morthisId))
+	if mess.VoiceState.UserID == morthisId {
+		sess.ChannelMessageSend(mess.ChannelID, fmt.Sprintf("Hello gaylord <@%s>", morthisId))
 	}
 
 	for _, griefee := range griefers {
-		if m.VoiceState.UserID == griefee {
-			s.GuildMemberMove(m.GuildID, griefee, &channelId)
+		if mess.VoiceState.UserID == griefee {
+			sess.GuildMemberMove(mess.GuildID, griefee, &channelId)
+		}
+	}
+}
+
+func birthdateCheck(sess *discordgo.Session) { 
+	var birthDateDiscId int
+	currDate := fmt.Sprintf(time.Now().Format("2006-01-02"))
+        targetTime := time.Date(now.Year(), now.Month(), now.Day(), 15, 0, 0, 0, now.Location()) 
+        if now.Hour() == targetTime.Hour() { 
+		birthDateDiscId = birthdateQuery(currDate)
+	}
+	birthDateDiscId = birthdateQuery(currDate)
+	if birthDateDiscId != -1 {
+		birthdayMessage := fmt.Sprintf("Happy Birthday: <@%d>", birthDateDiscId)
+
+		_, err := sess.ChannelMessageSend(tcTrainingId, birthdayMessage)
+		if err != nil {
+			fmt.Println("Error:", err)
+		
+		}
+	
+	
+	}
+}
+
+
+func birthdateQuery(dateToCheck string) int {
+	var discIdReturn int
+	myQuery := fmt.Sprintf("SELECT * FROM users WHERE TO_CHAR(dob, 'YYYY-MM-DD') LIKE '%%%s%%';", dateToCheck)
```

Comment:
This is a really hacky way to achieve what you want. It would be better to not have the `Dquery` wrapper function and instead to have a more controlled abstraction that lets us use the `*sql.DB` directly. This would allow for the following:

```go
rows, err := db.Query("SELECT id FROM users WHERE date_trunc('day', dob) = date_trunc('day', ?)")
if err != nil { ... }
defer rows.Close()

if !rows.Next() {
  return -1
}

var discordId int
err = rows.Scan(&discordId)
if err != nil { ... }
return discordId
```

Reply:
You can also achieve this by making `Dquery` pass arguments through which would look like the following:

```go
func Dquery(query string, args ...interface{}) (*sql.DB, *sql.Rows, error) {
   ...
  rows, err := db.Query(query, args...)
}